sunscreen_runtime/debugger-frontend/src/App.tsx

```diff
@@ -165,19 +202,58 @@ const dataToGraph = (data: FheProgramGraph, incRelin: boolean) => {
   return {nodes: nodes, edges: edges}
 }
 
+async function isProblematic(node, session: string) {
+  switch (node.type) {
+    case 'add':
+    case 'sub':
+    case 'multiply':
+      if (session.split('_')[0] === 'zkp') {
+        return false
+      } else {
+        const info = await fetch(`sessions/${session}/${node.id}`).then(d => d.json())
+        return info.Bfv.overflowed || info.Bfv.noise_budget <= 0
+      }
+    case 'constraint':
+      const info = await fetch(`sessions/${session}/${node.id}`).then(d => d.json())
+      return info.Zkp != 0
+    default:
+      return false;
+  }
+}
+
 
 const App = () => {
-  
-  const sampleData = JSON.parse(`{"graph":{"graph":{"nodes":[{"operation":{"InputCiphertext":0}},{"operation":{"InputCiphertext":1}},{"operation":{"InputCiphertext":2}},{"operation":{"InputCiphertext":3}},{"operation":"Multiply"},{"operation":"Multiply"},{"operation":"Multiply"},{"operation":"Add"},{"operation":"OutputCiphertext"},{"operation":"OutputCiphertext"},{"operation":"Relinearize"},{"operation":"Relinearize"},{"operation":"Relinearize"}],"node_holes":[],"edge_property":"directed","edges":[[0,4,"Left"],[3,4,"Right"],[1,5,"Left"],[2,5,"Right"],[1,6,"Left"],[3,6,"Right"],[12,7,"Left"],[10,7,"Right"],[7,8,"Unary"],[11,9,"Unary"],[5,10,"Unary"],[6,11,"Unary"],[4,12,"Unary"]]}},"data":"Bfv"}`)
-  const exGraph = dataToGraph(sampleData, true);
+
   const [selectedLine, setLine] = useState<number>(0);
   const [vertSize, setVertSize] = useState<any[]>();
   const [horSize, setHorSize] = useState<any[]>();
-  const [currGraph, setGraph] = useState(exGraph);
+  const [currCode, setCode] = useState<string>("select a session");
+  const [currGraph, setGraph] = useState({nodes: [], edges: []});
   const [selected, select] = useState<SelectionT | null>(null);
+  const [sessionList, setSessionList] = useState<string[]>([]);
+  const [session, setSession] = useState<string>("");
+  const [info, setInfo] = useState<any>({id: "no node selected"});
+  const [problemNodes, setProblemNodes] = useState<number[]>([]);
+
+  useEffect(
+    () => {fetch("/sessions").then(j => j.json()).then(l => setSessionList(l))}, []
+  )
+
+  const updateProblematicNodes = useCallback( async (graph) => {
```

Comment:
I don't think your current approach works the way you think it does. You are caching a function `updateProblematicNodes` based on a graph, but then you always pass in a newly fetched graph when it is called below. This should just be a normal function, with no arguments, that refers to the `currGraph` already in scope.